database-rules.json

```diff
@@ -199,6 +199,14 @@
         }
       }
     },
+    // Privacy settings
+    "privacy": {
+      "$uid": {
+        ".write": "auth.uid === $uid",
+        ".read": "auth.uid === $uid",
+        ".validate": "newData.child('data_processing').isBoolean() && newData.child('content').isBoolean() && newData.child('social').isBoolean()"
```

Comment:
Let's make this a bit better and disallow writing random other data to the database:

```json
"privacy": {
  "$uid": {
    ".write": "auth.uid === $uid",
    ".read": "auth.uid === $uid",
    "data_processing": {
      ".validate": "newData.child('data_processing').isBoolean()"
    },
    "content": {
      ".validate": "newData.child('content').isBoolean()"
    },
    "social": {
      ".validate": "newData.child('social').isBoolean()"
    },
    "$other": {
      ".validate": "false"
    }
  }
},
```